### PR TITLE
Add tests for all enumerable methods

### PR DESCRIPTION
**Pull request by HillaryOkello**
In this branch, I created the test for the enumerable methods using RSpec.

### CODE REVIEW

**Comment by descholar-ceo** [comment]
# This Pull Request is under review by [descholar-ceo](https://github.com/descholar-ceo) :eyes: :eyes: 
## Please wait a bit ...

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @HillaryOkello  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
#### :warning: However we need to make few changes to mark this project approved. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [ ] Missing test cases generally for all methods:
    * Test when there is no block given (how the method behaves)
- [ ] For `#my_each` and `#my_each_with_index` test also when self is a hash
- [ ] For `#my_all?`, `#my_any?`, `#my_none?` test also the following test cases:
    * if the block never returns false or nil
    * if the block returns false or nil
    * if no block or argument is given
    * if one of the collection members are false or nil
- [ ] Add tests for `#multiply_els` method
- [ ] Update the readme file and add how to run tests
- [ ] Fix rubocop lint issues

# [Optional changes] :yellow_circle: 
:warning: Your commit messages should be in imperative mode or present tense, _i.e.:_ you should have written `Modified enum_spec: add more rules from line 12 to line 16 ` instead of writing `modified enum_spec`, refer [to this article for more details on how to write a good && professional commit message](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

**Comment by descholar-ceo** [comment]
# This Pull Request is under review by [descholar-ceo](https://github.com/descholar-ceo) :eyes: :eyes: 
## Please wait a bit ...

File: spec/enum_spec.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  describe '#my_each' do
+    it 'Adding 1 to each array item' do
+      array = []
+      [1, 2, 3].my_each { |x| array.push(x + 1) }
+      expect(array).to eql([2, 3, 4])
+    end
+
+    it "my_each iteration testing with no block" do
+      expect(arr.my_each {|i|}).to eql(arr.each {|i|})
+    end
```
Comment (descholar-ceo) [inline spec/enum_spec.rb]: To test the `#my_each` method when no block passed is not done like this you are doing here. You should run something like `arr.my_each` :point_left: That way there is no block passed, but when you give it anything in curly blaces then that is a block you are passing it.

Hint: the original `#each` method returns an enumerator if there is no block, so write a test to check if your `#my_each` method returns an enumerator. The same hint applies on `#my_each_with_index`

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @HillaryOkello  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implemented the requested changes from the previous review, keep it up, and your project is really almost done.
#### :warning: However we need to make few changes to mark this project approved, don't feel discouraged because of being asked some changes more and more, that's beacuse this is a learning process, and we are here to help you to become an out-standing developer. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [ ] You did not implement all of the changes I requested on the first review, please implement all of the required changes before you submit for the other re-review
___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Request changes 😳

I am very happy to see this project and I can see you worked hard on this project. There are only a few issues that we need to address and I am sure that you will end up with a perfect project

_It's okay. We are here to learn. Don't get discouraged. Work even harder. You are almost there.😄_

This is a great opportunity to learn. Microverse is here to help you. 

**Please have a look at the suggestions to get more understanding.**

- [ ] Please remove all Rspec errors. Always confirm that you have no errors before the code review. 


**Please make the changes and request a re-review for the given project**

If you get stuck somewhere, don't hesitate to reach out or schedule s 1:1 meeting with a Full-time TSE. 

Stay safe and give me a smile 😇

### Happy Coding
---------------------------------------------------------------------------------------------------------------------------------
**Regards,
Arslan Bisharat
slack: arslanbisharat**

**Review by PhillipUg — APPROVED** [review]
# Status: Project Approved! 🥇 
Congratulations on implementing all requirements and getting this project approved. :+1:

You can merge this branch into the development branch when you're ready and go on to your next project.
:bulb:  Do not forget to submit your project completion form after merging this branch.

All the best team,

For any assistance, you can contact me on slack as **Phillip Musiime**

**Pull request merged**